src/main/java/controller/Controller.java

```diff
@@ -0,0 +1,61 @@
+package controller;
+
+import java.util.ArrayList;
+import java.util.List;
+import model.Car;
+import model.Cars;
+import util.generator.NameGenerator;
+import util.generator.TrialNumberGenerator;
+import util.validator.TrailNumberValidator;
+import view.InputView;
+import view.OutputView;
+
+public class Controller {
+	private Cars cars;
+	private int trialNumber;
+
+	public void run(){
+		initialize();
+		OutputView.printMessage();
+		moveCarByTrialNumber();
+		OutputView.printWinner(cars.findWinners());
+	}
+
+	private void initialize(){
+		cars = new Cars(getCars());
+		trialNumber = getTrialNumber();
+	}
+
+	private List<Car> getCars(){
+		List<Car> car = new ArrayList<>();
+		try{
+			List<String> names = NameGenerator.generatorCarsName(InputView.getCarNames());
+			addNewCars(car, names);
+			return car;
+		} catch (IllegalArgumentException e){
+			return getCars();
+		}
+	}
+
+	private void addNewCars(List<Car> cars, List<String> names){
+		for(String name : names){
+			cars.add(new Car(name));
+		}
+	}
+
+	private int getTrialNumber(){
+		String input = InputView.getTrialNumber();
+		try{
+			return TrialNumberGenerator.generateTrialNumber(input);
+		} catch (IllegalArgumentException e){
+			return getTrialNumber();
+		}
+	}
```

Comment:
```suggestion
	private int getTrialNumber() throw IllegalArgumentExcetion {
		String input = InputView.getTrialNumber();
			return TrialNumberGenerator.generateTrialNumber(input);
	}
```

이렇게 바꾸고 최상단에서 try-catch로 전부 잡아내는 방법도 있어요.
이걸 checked exception, unchecked exception으로 분리해서 나눠 볼 수 있는데,
자세한 이야기는 [예외 처리 전략](https://www.nextree.co.kr/p3239/)에 대해 찾아보면 좋을 것 같네요.
정답은 없으니 본인 취향에 맞게 고쳐보면 좋을 것 같습니다.